ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/sandbox/services/SandboxFixtureWithAuth.scala

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.platform.sandbox.services
+
+import java.time.{Duration, Instant}
+import java.util.UUID
+
+import com.digitalasset.grpc.{GrpcException, GrpcStatus}
+import com.digitalasset.jwt.domain.DecodedJwt
+import com.digitalasset.jwt.{HMAC256Verifier, JwtSigner}
+import com.digitalasset.ledger.api.auth.{AuthServiceJWT, AuthServiceJWTCodec, AuthServiceJWTPayload}
+import com.digitalasset.ledger.api.v1.ledger_offset.LedgerOffset
+import com.digitalasset.ledger.api.v1.transaction_filter.{Filters, TransactionFilter}
+import com.digitalasset.ledger.client.auth.LedgerClientCallCredentials.authenticatingStub
+import com.digitalasset.platform.sandbox.config.SandboxConfig
+import com.digitalasset.platform.testing.SingleItemObserver
+import io.grpc.Status
+import io.grpc.stub.{AbstractStub, StreamObserver}
+import org.scalatest.Suite
+import scalaz.syntax.tag.ToTagOps
+
+import scala.concurrent.{ExecutionContext, Future, Promise}
+
+trait SandboxFixtureWithAuth extends SandboxFixture { self: Suite =>
+
+  protected def stub[A <: AbstractStub[A]](stub: A, token: Option[String]): A =
+    token.fold(stub)(authenticatingStub(stub))
+
+  override protected def config: SandboxConfig =
+    super.config.copy(
+      authService = Some(
+        AuthServiceJWT(HMAC256Verifier(jwtSecret)
+          .getOrElse(sys.error("Failed to create HMAC256 verifier")))))
+
+  protected def roToken(party: String) = AuthServiceJWTPayload(
+    ledgerId = None,
+    participantId = None,
+    applicationId = None,
+    exp = None,
+    admin = false,
+    actAs = Nil,
+    readAs = List(party)
+  )
+
+  protected def rwToken(party: String) = AuthServiceJWTPayload(
+    ledgerId = None,
+    participantId = None,
+    applicationId = None,
+    exp = None,
+    admin = false,
+    actAs = List(party),
+    readAs = List(party)
+  )
+
+  protected def adminToken = AuthServiceJWTPayload(
+    ledgerId = None,
+    participantId = None,
+    applicationId = None,
+    exp = None,
+    admin = true,
+    actAs = List(),
+    readAs = List()
+  )
+
+  protected lazy val wrappedLedgerId = ledgerId(Some(adminToken.asHeader()))
+  protected lazy val unwrappedLedgerId = wrappedLedgerId.unwrap
+
+  private val jwtHeader = """{"alg": "HS256", "typ": "JWT"}"""
+  private val jwtSecret = UUID.randomUUID.toString
+
+  implicit protected class AuthServiceJWTPayloadExtensions(payload: AuthServiceJWTPayload) {
+    def expiresIn(t: java.time.Duration): AuthServiceJWTPayload =
```

Comment:
From a recent PR (https://github.com/digital-asset/daml/pull/3548/files#diff-f00a01a83ce733d3a2bf7e966718eb3dR105):
```
    def readonly(): AuthServiceJWTPayload =
      payload.copy(readAs = payload.actAs ++ payload.readAs, actAs = List.empty)
```

Would replace the need for `rwToken` and `roToken`.